staging/src/k8s.io/apiserver/pkg/server/filters/timeout.go

```diff
@@ -97,17 +98,31 @@ func (t *timeoutHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	tw := newTimeoutWriter(w)
 	go func() {
 		defer func() {
-			err := recover()
+			rval := recover()
 			// do not wrap the sentinel ErrAbortHandler panic value
-			if err != nil && err != http.ErrAbortHandler {
+			if rval != nil && rval != http.ErrAbortHandler {
 				// Same as stdlib http server code. Manually allocate stack
 				// trace buffer size to prevent excessively large logs
 				const size = 64 << 10
 				buf := make([]byte, size)
 				buf = buf[:runtime.Stack(buf, false)]
-				err = fmt.Sprintf("%v\n%s", err, buf)
+				rval = fmt.Sprintf("%v\n%s", rval, buf)
+			}
+			// resultCh will have no reader if the enclosing function
+			// has already exited.
+			select {
+			case resultCh <- rval:
```

Comment:
actually, I think this introduces a race in the opposite direction... execution can reach here prior to the `select` between the result and timeout channels, making this take the default case and leaving that select to timeout

Reply:
ah maybe that explains why all the integration tests are failing.